core/src/wallet/tezos/TezosLikeAccount2.cpp

```diff
@@ -369,15 +369,34 @@ namespace ledger {
+                                            }).flatMapPtr<TezosLikeTransactionApi>(self->getMainExecutionContext(), [self, explorer, tx, senderAddress] (const std::shared_ptr<Block> &block) {
                                                 tx->setBlockHash(block->hash);
                                                 if (senderAddress.find("KT1") == 0) {
                                                     // HACK: KT Operation we use forge endpoint
-                                                    return explorer->forgeKTOperation(tx).mapPtr<api::TezosLikeTransaction>(self->getMainExecutionContext(), [tx] (const std::vector<uint8_t> &rawTx) {
+                                                    return explorer->forgeKTOperation(tx).mapPtr<TezosLikeTransactionApi>(self->getMainExecutionContext(), [tx] (const std::vector<uint8_t> &rawTx) {
                                                         tx->setRawTx(rawTx);
                                                         return tx;
                                                     });
                                                 }
+                                                return FuturePtr<TezosLikeTransactionApi>::successful(tx);
+                                                }).flatMapPtr<api::TezosLikeTransaction>(self->getMainExecutionContext(), [self, request] (const std::shared_ptr<TezosLikeTransactionApi> &tx) {
+                                                        if (request.gasLimit->toInt() == 0) {
+                                                            auto filledTx = tx;
+                                                            auto gasPrice_fut = request.fees->toInt() == 0 ?
+                                                                self->getGasPrice()
+                                                                :
+                                                                FuturePtr<BigInt>::successful(request.fees);
+
+                                                            return gasPrice_fut.flatMapPtr<api::TezosLikeTransaction>(self->getMainExecutionContext(), [self, filledTx] (const std::shared_ptr<BigInt>&gasPrice) -> FuturePtr<api::TezosLikeTransaction> {
+                                                                return self->estimateGasLimit(filledTx).flatMapPtr<api::TezosLikeTransaction>(self->getMainExecutionContext(), [filledTx, gasPrice] (const std::shared_ptr<BigInt> &gas) -> FuturePtr<api::TezosLikeTransaction> {
+                                                                    // 0.000001 comes from the gasPrice being in picoTez
+                                                                    const auto fees = std::make_shared<BigInt>(static_cast<int64_t>(1 + gas->toInt64() * static_cast<double>(gasPrice->toInt64()) * 0.000001));
```

Comment:
+1 is to get the `ceil` of the value instead of the `floor`, and I need to cast the gasPrice to double to make sure that multiplication by `0.000001` is not truncated to 0. I'll cast everything inside